crates/task-impls/src/transactions.rs

```diff
@@ -219,24 +216,32 @@ where
                 // TODO (Keyao) Determine whether to allow empty blocks.
                 // <https://github.com/EspressoSystems/HotShot/issues/1822>
                 let txns = self.wait_for_transactions(parent_leaf).await?;
-                let encoded_txns = VIDTransaction::encode(txns.clone());
+                // TODO (Keyao) We encode the transactions and compute the VID disperse data twice,
```

Comment:
I'm not sure this is actually redundant. `from_transactions` and `encode` are doing semantically different things: `from_transactions` is converting a list of transactions into an application-specific block type, which may be represented a variety of ways. `encode` is requesting a specific representation (flat bytes) of that payload.

In practice, in the sequencer, we plan to have our implementation of `BlockPayload` store the flat bytes directly, since we need it anyways for doing VID proofs. Then `from_transactions` will do the work of flattening, and `encode` will just return the flat bytes, so no work is wasted.

Reply:
Makes sense! How about the VID disperse calculation? I think both `from_transactions` (in the sequencer implementation) and `vid.disperse` will do it.

Reply:
I think we need to clarify. The sequencer (ie. the impl of `BlockPayload`) will never call VID `disperse` or `commit_only`. It's the job of hotshot to compute the VID commitment when needed. (The leader gets the commitment from `disperse`, the DA committee needs to call `commit_only` to verify the payload bytes.)

Reply:
Got it, thanks for the clarification. Then there's no redundancy here.